### PR TITLE
Updated Styled Components to 4.0.2, Babel to 7.1.2

### PR DESCRIPTION
**Pull request by timhagn**
First of all, great module, I really like using it with Styled Components!

In preparation for writing the tests with jest, it came to me, that some dependencies were
quite outdated - especially now that styled-components is @ 4.0.2. 
So I did the following:

- [x] updated styled-components ^3.1.5 to ^4.0.2
- [x] babel-preset-env@^1.6.0 to @babel/core@^7.1.2 & @babel/preset-env@^7.1.0
- [x] updated rollup ^0.45.2 to ^0.66.6 - and it's plugins accordingly
- [x] changed rollup.config.js to newest config styles (no warnings with yarn build anymore)
- [x] removed .babelrc in favor of .babelrc.js

And some smaller changes, as you can see in the commits.

**With all this I wanted to:**
1. Bump all to the latest versions
2. Prepare for writing the tests in jest

I tested the build process and integrated it in a gatsby page with no problems occuring.

**This will prepare for:**
Tackling Issue #1 - If it's at your convenience.

### CODE REVIEW

**Comment by morajabi** [comment]
It looks good! I will try to merge this in later today! Are we sure with peerDep versions? Many people aren't at latest React and might get a little warning from NPM / Yarn cli.

**Comment by timhagn** [comment]
Thanks : ). And if they want to use the latest styled-components, they should be, cause there react / react-dom is a dev-Dependency: https://github.com/styled-components/styled-components/blob/master/package.json - so I guess, we are in the clear.

**Comment by aureliome** [comment]
Good job @timhagn , I'm waiting this merge with trepidation :)

**Comment by timhagn** [comment]
@aureliome Thanks : )! Did you choose the word "trepidation" for Halloween? Cause translated to German it would mean something like "anxiety", "fearfulness" or the like ^^. One never stops learning.
But @morajabi, did you already get the chance to look into merging it (or the tests in #14, for what it's worth)? I'm already using the updated versions in a soon to be production environment with no issues
(putting `"styled-media-query": "https://github.com/timhagn/styled-media-query#th-jest-tests"` in my package.json for the time being ; ).

**Comment by franky47** [comment]
I'm really interested in this merge as well ! I'm using `styled-components` v4 with TypeScript, I'll give this branch a go in the mean time.

File: .babelrc.js
Comment (ApacheEx) [inline .babelrc.js]: let's add a newline at the end of file

File: .gitignore
Comment (ApacheEx) [inline .gitignore]: I think such things should be handled in global .gitignore instead of this library.

File: package.json
Comment (ApacheEx) [inline package.json]: Do not see any reason to remove support for styled-components < 4.

So, let's have `">= 1"` here.

File: package.json
Comment (ApacheEx) [inline package.json]: `styled-media-query` doesn't depend on React and I think we should not require people to upgrade to avoid warnings when install this library.

**Review by ApacheEx — CHANGES_REQUESTED** [review]
1. First of all, you did a great job! Thank you. 🥇 

2. I have a few remarks - hope it's clear, but feel free to discuss 👍 

3. p.s. do we need package-lock.json here? I think yarn.lock is enough. ❓

File: package.json
Comment (timhagn) [reply package.json]: See my above comment: react & react-dom are meanwhile devDependency in styled-components. And it's needed for the tests anyhow...

File: package.json
Comment (morajabi) [reply package.json]: I agree here.

File: .gitignore
Comment (morajabi) [reply .gitignore]: I think @timhagn uses IntelliJ and they wanted to ignore .idea files :)

**Comment by morajabi** [comment]
@timhagn Could you please change the version range like @ApacheEx commented above? Otherwise LGTM :)

File: package.json
Comment (ApacheEx) [reply package.json]: @timhagn, you mixed 2 PRs and it's not easy to review. I think this PR should not include any changes for tests which are done in #15.

**Comment by timhagn** [comment]
For all waiting for an answer, my main workstations Motherboard crashed yesterday and I'm just trying to get back on track with my Dev-Laptop -.-
Sorry for the inconvenience, will try to resolve the requested changes as soon as possible % ).

File: .gitignore
Comment (timhagn) [reply .gitignore]: *g* I usually have em in a .gitignore_global - maybe I've got to review my settings for the .ignore plugin ^^. Will remove it.

File: package.json
Comment (timhagn) [reply package.json]: Will do, as soon as I'm up and running again (tomorrow / day after tomorrow).

File: package.json
Comment (timhagn) [reply package.json]: OK, I concede ^^.

**Comment by morajabi** [comment]
No worries @timhagn! It's open source and we're here to learn and have fun 👋 hope the machine gets alright soon!

File: .babelrc.js
Comment (timhagn) [reply .babelrc.js]: #done

File: .gitignore
Comment (timhagn) [reply .gitignore]: #removed

File: package.json
Comment (timhagn) [reply package.json]: #done

File: package.json
Comment (timhagn) [reply package.json]: #done

**Comment by timhagn** [comment]
Although my main workstation is still down:
Finally! Finalmente! Endlich!
Squashed all changes done before into one commit and integrated all your suggestions.
LGTM : )!
(Used my Dev-Laptop ; ).

**Review by ApacheEx — APPROVED** [review]

**Comment by ApacheEx** [comment]
I have removed `package-lock.json` and also unnecessary blank line in `.babelrc.json`, now it looks good for me. Thank you a lot 🥇

p.s. @morajabi I think it's ready to merge

**Comment by morajabi** [comment]
@ApacheEx Nice!
@timhagn Thanks a lot for the hard work, I'm excited to merge and release this :D

**Comment by timhagn** [comment]
@ApacheEx Thanks for these changes! I'm only irritated a bit about the removal of the "unnecessary blank line, as I added it upon your [request](https://github.com/morajabi/styled-media-query/pull/14#discussion_r234411931) ^^.
@morajabi  Thanks, was a pleasure : ). As I'm using styled-media-query to ease my life right now, I'd have some ideas on API v3 - but LGTM first ; ).

**Comment by timhagn** [comment]
Any updates on this merge? Just asking ; ).

**Comment by morajabi** [comment]
First, my apologies for the merging taking so long! It's now published as v2.1.0 🎷
Second, thanks Tim for the PRs, I just invited you to the repository as a collaborator. Although you already have helped with the project, I'd like to have you as a collaborator. This means you can review and manage PRs or issues or even merge PRs when you or the collaborators, or I agree on.

**Comment by timhagn** [comment]
Nice! That's greats news on the merge and me now being a contributor. Thanks a lot, gonna use it responsibly : ). And so happy, that I now don't have to integrate my repo in my package.json ^^.

**Comment by ApacheEx** [comment]
@morajabi,

I get this error after upgrading, seems github release is missing.
image

**Comment by morajabi** [comment]
@ApacheEx Forgot to push the GH tags, but how does that make the build fail? I thought NPM doesn't depend on these.

**Comment by ApacheEx** [comment]
I'm still faced this issue, is it only me?
image

**Comment by timhagn** [comment]
Nope, it isn't only you, @ApacheEx, I just tried it and getting the same result -.-
Shouldn't there be a dist folder in the npm package, or how do you finalize the
build process, @morajabi? Hope I didn't break it % ).

**Comment by morajabi** [comment]
Wait, I think I found the cause.

**Comment by morajabi** [comment]
@timhagn @ApacheEx it should be fixed 🙈

**Comment by ApacheEx** [comment]
yeah, works now. Thank you guys

**Comment by timhagn** [comment]
@morajabi @ApacheEx Yup 2.1.1 is working and has a `dist` folder ^^. Thanks!
P.S.: What was the problem's source, if you'd indulge, @morajabi?

**Comment by morajabi** [comment]
@timhagn Someone forgot to run `yarn build` before publishing 😅 (me)

**Pull request merged**
